test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1

```diff
@@ -1,230 +1,732 @@
+            Move-Item -Path "TestDrive:\*" -Destination $altTestFile -Include "*.txt" -Exclude "*2*"
+            $file1 = Get-Item $testFile2 -ErrorAction SilentlyContinue
+            $file2 = Get-Item $altTestFile -ErrorAction SilentlyContinue
+            $file1 | Should BeNullOrEmpty
+            $file2.Name | Should Be $altTestFile
+        }
+
+        It "Verify Filter" -Skip:$true { #Skipped until issue #2385 gets resolved
+            Move-Item -Path "TestDrive:\*" -Filter "*2.txt" -Destination $altTestFile
+            $file1 = Get-Item $testFile2 -ErrorAction SilentlyContinue
+            $file2 = Get-Item $altTestFile -ErrorAction SilentlyContinue
+            $file1 | Should BeNullOrEmpty
+            $file2.Name | Should Be $altTestFile
+        }
     }
 
-    It "Set-Content to a file" {
-        $content =  Set-Content -Value "some content" -Path $testFile -PassThru
-        $content | Should BeExactly "some content"
-    } 
+    Context "Valdiate Rename-Item parameters for the FileSystem provider" {
+        BeforeAll {
+            $newFile = "NewName.txt"
+        }
+
+        BeforeEach {
+            New-Item -Path $testFile -ItemType File > $null
+        }
+
+        AfterEach {
+            Remove-Item -Path * -Recurse -Force -ErrorAction SilentlyContinue
+        }
+
+        It "Verify WhatIf" {
+            Rename-Item -Path $testFile -NewName $newFile -WhatIf
+            try {
+                Get-Item -Path $newFile -ErrorAction Stop 
+                throw "Expected exception not thrown"
+            }
+            catch { $_.FullyQualifiedErrorId | Should Be "PathNotFound,Microsoft.PowerShell.Commands.GetItemCommand" }
+        }
 
-    It "Add-Content to a file" {
-        $content = Set-Content -Value "some content" -Path $testFile -PassThru
-        $addContent = Add-Content -Value " more content" -Path $testFile -PassThru
-        $fullContent = Get-Content -Path $testFile
-        $content | Should Match "some content"
-        $addContent | Should Match "more content"
-        ($fullContent[0] + $fullContent[1]) | Should Match "some content more content"
+        It "Verify Confirm can be bypassed" {
+            Rename-Item -Path $testFile -NewName $newFile -Confirm:$false
+            $file1 = Get-Item -Path $testFile -ErrorAction SilentlyContinue
+            $file2 = Get-Item -Path $newFile -ErrorAction SilentlyContinue
+            $file1 | Should BeNullOrEmpty
+            $file2.Name | Should Be $newFile
+        }
     }
 
-    It "Clear-Content of a file" {
-        Set-Content -Value "some content" -Path $testFile
-        $contentBefore = Get-Content -Path $testFile
-        Clear-Content -Path $testFile
-        $contentAfter = Get-Content -Path $testFile
-        $contentBefore.Count | Should Be 1
-        $contentAfter.Count | Should Be 0
+    Context "Valdiate Remove-Item parameters for the FileSystem provider" {
```

Comment:
You are right. I factored out all of the navigation tests into a new describe block called Describe "Extended FileSystem Path/Location Cmdlet Provider Tests".